code-challenges.test.js

```diff
@@ -63,32 +99,79 @@ var randomNouns2 = ["temperature", "database", "chopsticks", "mango", "deduction
+    const sortedTwo = numbersLeastToGreat(preSortedTwo)
+    expect (sortedTwo).toEqual([-9, 0, 3, 18, 94])
+  });
+    
+})
+
+// // // b) Create the function that makes the test pass.
+
+// // we want a function that will that will decide on each valur in our array for the type of being a number; therefore, ...
+// // we will use .filter and select for "number"
+// tried to use .map and .push after to rearrange the array but couldn't get the logic correct... therefore, learned of .sort() which can be used with function expressions to re-arrange the array. However, our array stores the numbers as strings so we have to implement a compare function. Without the compare function the numbers would be sorted as if 18 is smaller than 3 as 1 is smaller than 3. Therefore, the compare function allows us to define an alternate sort order than what is built in. It works by comparing two values at a time (a and b). The method calls the compare function to compare A and B. The function calculates A minus B and if A is less than B the result will be negative indicating A being a smaller value than B and it will be sorted in that method.
+
+const numbersLeastToGreat = (containsFiltered) => {
+    let filteredArray = containsFiltered.filter( numbers => {
+        return (typeof numbers === "number")
+    });
+    return filteredArray.sort(function(a,b){
+        return a-b;
+    })
+}
+//.sort( is )
+//.map only iterates in order?
+//.push
+
+// // --------------------4) Create a function that takes in a string and logs the index of the first vowel.
+
+// // a) Create a test with expect statements for each of the variables provided.
 
 var vowelTester1 = "learn"
-// Expected output: 1
+// // // Expected output: 1
 var vowelTester2 = "academy"
-// Expected output: 0
+// // // Expected output: 0
 var vowelTester3 = "challenge"
-// Expected output: 2
+// // // Expected output: 2
+
+describe ("When 'vowelIndexer' is called", () => {
+  it ("When 'learn' is tested return '1'", () => {
+    const word1 = "learn"
+    const foundVowel1 = vowelIndexer (word1)
+    expect (foundVowel1).toEqual(1)
+  })
+  it ("When 'academy' is tested return '0'", () => {
+    const word2 = "academy"
+    const foundVowel2 = vowelIndexer (word2)
+    expect (foundVowel2).toEqual(0)
+  })
+  it ("When 'learn' is tested return '2'", () => {
+    const word3 = "challenge"
+    const foundVowel3 = vowelIndexer (word3)
+    expect (foundVowel3).toEqual(2)
+  }) 
+})
 
 
 
+// b) Create the function that makes the test pass.
 
+// we want a function that will index our string and indicate the index of the first occurance of a vowel
+// we could use regular expression (way easier) or convert our string (way harder) into an indexable array and iterate over each value to return the first instance of a vowel.
 
-// b) Create the function that makes the test pass.
+const vowelIndexer = (str) => {
+    return str.search(/[a,e,i,o,u]/i)
```

Comment:
using regex is fine but it's an intense tool for a simple job. It also requires other developers to know how to read regex. Ideally, when you use regex, you'll also want to provide a comment about what it is doing

```suggestion
    return str.search(/[aeiou]/i)
```
Commas within this regex are not needed because the way this works is the search will look at the regex and look at each character within the square brackets and treat each character as a character it must look for. The commas are not something we're looking for in the string (just vowels) and if it were, we'd only need to include it once